### PR TITLE
Issue 522 dream

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: src/modules/dream/readout/DataParser.cpp [hunk 1]
```diff
@@ -10,12 +10,11 @@
 #include <common/readout/ess/Parser.h>
 #include <dream/readout/DataParser.h>
 
-// #undef TRC_LEVEL
-// #define TRC_LEVEL TRC_L_DEB
+#undef TRC_LEVEL
+#define TRC_LEVEL TRC_L_WAR
```
Comment (JennyLouise) [inline src/modules/dream/readout/DataParser.cpp:14]: Debug messages left in

File: src/modules/dream/readout/DataParser.cpp [hunk 2]
```diff
@@ -29,67 +28,59 @@ int DataParser::parse(const char *Buffer, unsigned int Size) {
...
-    auto DataHdrPtr = (ESSReadout::Parser::DataHeader *)DataPtr;
+    auto Data = (DreamReadout *)((char *)DataPtr);
 
-    if (BytesLeft < DataHdrPtr->DataLength) {
-      XTRACE(DATA, DEB, "Data size mismatch, header says %u got %d",
-             DataHdrPtr->DataLength, BytesLeft);
-      Stats.ErrorHeaders++;
+    ///\todo clarify distinction between logical and physical rings
+    // for now just divide by two
+    Data->RingId = Data->RingId / 2;
```
Comment (JennyLouise) [inline src/modules/dream/readout/DataParser.cpp:41]: We need to decide where this happens - some modules /2 in the data parser, others in the instrument

File: src/modules/dream/readout/DataParser.cpp [hunk 3]
```diff
@@ -29,67 +28,59 @@ int DataParser::parse(const char *Buffer, unsigned int Size) {
...
+
+    Result.push_back(*Data);
+    BytesLeft -= Data->DataLength;
+    DataPtr += Data->DataLength;
   }
 
   return ParsedReadouts;
 }
-} // namespace Dream
+} // namespace Loki
```
Comment (JennyLouise) [inline src/modules/dream/readout/DataParser.cpp:86]: Wrong namespace

File: src/modules/dream/test/CuboidTest.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  ASSERT_EQ(x,  1);
+  ASSERT_EQ(y, 12);
+}
+
+TEST_F(CDTGeometryTest, Rotate1) {
+  int x = 1;
+  int y = 12;
+  geometry.rotateXY(x, y, 1);
+  ASSERT_NE(x,  1);
+  ASSERT_NE(y, 12);
```
Comment (JennyLouise) [inline src/modules/dream/test/CuboidTest.cpp:36]: Does ASSERT_NE check thoroughly enough? Can we calculate what the rotated values should be and use ASSERT_EQ instead?

File: src/modules/dream/test/DreamBaseTest.cpp [hunk 5]
```diff
@@ -66,16 +70,17 @@ std::vector<uint8_t> TestPacket2{
...
+    0x01, 0x00, 0x00, 0x00, // Seq number 1
 
     // Data Header 1
     0x00, 0x01, 0x10, 0x00, // ring 0, fen 1, data size 16 bytes
 
     // Readout 1
     0x22, 0x00, 0x00, 0x00, // tof 34 (0x22)
     0x00, 0x00, 0x14, 0x05, // unused 00 00 module 20, sumo 5
-    0x0e, 0x0b, 0x01, 0x01  // strip 14, wire 11, segment 1, counter 2
+    0x00, 0x00, 0xCC, 0xAA  // strip 14, wire 11, segment 1, counter 2
```
Comment (JennyLouise) [inline src/modules/dream/test/DreamBaseTest.cpp:83]: The hex values changed but the values in the comment didn't, are they out of date now?

**Review by JennyLouise — APPROVED** [review]
I made some small suggestions in the comments, but nothing major, looks good to me.

File: src/modules/dream/readout/DataParser.cpp (hunk 2 above)
Comment (mortenjc) [reply src/modules/dream/readout/DataParser.cpp:41]: Agree. I would say as early as possible in general.

**Pull request merged**